Basic_viewer/examples/Basic_viewer/draw_surface_mesh_height.cpp

```diff
@@ -0,0 +1,67 @@
+#include <CGAL/Simple_cartesian.h>
+#include <CGAL/Surface_mesh.h>
+#include <CGAL/draw_surface_mesh.h>
+#include <CGAL/Drawing_functor.h>
+#include <fstream>
+
+typedef CGAL::Simple_cartesian<double>                       Kernel;
+typedef Kernel::Point_3                                      Point;
+typedef CGAL::Surface_mesh<Point>                            Mesh;
+
+struct Colored_faces_given_height: public CGAL::Drawing_functor<Mesh,
+                                   typename Mesh::Vertex_index,
+                                   typename Mesh::Edge_index,
+                                   typename Mesh::Face_index>
+{
+  Colored_faces_given_height(const Mesh& sm)
+  {
+    double m_min_z, m_max_z;
+    if(!sm.is_empty())
+    {
+      bool first=true;
+      for(typename Mesh::Vertex_index vi: sm.vertices())
+      {
+        if(first)
+        { m_min_z=sm.point(vi).z(); m_max_z=m_min_z; first=false; }
+        else
+        {
+          m_min_z=std::min(m_min_z, sm.point(vi).z());
+          m_max_z=std::max(m_max_z, sm.point(vi).z());
+        }
+      }
+    }
+
+    this->colored_face=[](const Mesh &, typename Mesh::Face_index)->bool { return true; };
+
+    this->face_color=[m_min_z, m_max_z]
+      (const Mesh& sm, typename Mesh::Face_index fi)->CGAL::IO::Color
+    {
+      double res;
+      std::size_t n=0;
+      for(typename Mesh::Vertex_index vi: vertices_around_face(sm.halfedge(fi), sm))
+      {
+        res+=sm.point(vi).z();
+        ++n;
+      }
+      // Random color depending on the "height" of the facet
```

Comment:
Why random?    Would it not be be more reasonable to have a color that depends on the height, ideally a gradient?

Reply:
Indeed this is better (but an implementation detail for this example)